test/integration/apiserver/watchcache_test.go

```diff
@@ -234,3 +237,54 @@ func BenchmarkListFromWatchCache(b *testing.B) {
 		b.Logf("Number of secrets: %d", len(secrets.Items))
 	}
 }
+
+func TestList(t *testing.T) {
```

Comment:
As discussed offline, I think we can do better job on testing the feature by using very high periodic bookmark interval, requests on two resource types.

Reply:
Done.